File: server/lib/PointCounter.js
export {
    countPointScore
};


function countPointScore(cards = [], picture, ace, number) {
    return cards.reduce((accumulator, card) =>
        accumulator + (
            (card % 13 == 0) ? ace :
                (card % 13 <= 12 && card % 13 >= 10)
                    ? picture : number), 0);
}

File: server-rust/src/helpers/point_counter.rs

pub fn count_point_score(cards: &Vec<u32>, picture: u32, ace: u32, number: u32) -> u32 {
    cards.iter().fold(0, |accumulator, &card| {
        accumulator
            + if card % 13 == 0 {
                ace
            } else if card % 13 <= 12 && card % 13 >= 10 {
                picture
            } else {
                number
            }
    })
}

